syntax = "proto3";
option cc_enable_arenas = true;

package NFq.NConfig;
option java_package = "ru.yandex.kikimr.proto";

////////////////////////////////////////////////////////////

message TYdbDriverConfig {
    uint64 NetworkThreadsNum = 1; // default: 2
    uint64 ClientThreadsNum = 2; // default: 0 that means adaptive thread pool
    uint64 GrpcMemoryQuota = 3; // default: 0 (unlimited)
    uint32 MonitoringPort = 4; // default: 0 (disabled)
}

message TCommonConfig {
    TYdbDriverConfig YdbDriverConfig = 1;
    string YdbMvpCloudEndpoint = 2;
    string MdbGateway = 3;
    bool UseBearerForYdb = 4;
    bool MdbTransformHost = 5;
    string ObjectStorageEndpoint = 6;
    string IdsPrefix = 7;
    uint64 MaxTasksPerOperation = 8;
    string QueryArtifactsCompressionMethod = 9;
    uint64 QueryArtifactsCompressionMinSize = 10;
    string MonitoringEndpoint = 11;
    uint64 MaxTasksPerStage = 12;
    bool KeepInternalErrors = 13;
    bool UseNativeProtocolForClickHouse = 14;
    bool DisableSslForGenericDataSources = 15;
    bool ShowQueryTimeline = 16;
}
